### ISSUE TITLE
wrong Limit when rate > 1

### ISSUE THREAD

**Issue by july2993** [bsm/ratelimit#1]
current := atomic.AddUint64(&rl.allowance, passed)
i thing this line shoud be the following line for rl.rate be the rate (pass by the New function)
current := atomic.AddUint64(&rl.allowance, passed*uint64(rl.rate))

-------
New(100,  time.Second)
consider this limiter, allowance will be rate*time.Second first, but it will only be added most time.Second per second, but one call Limit consume allowance time.Second. so if you call Limit() quickly, first you can call about 100 time in the first second, then only one per second.

**Comment by dim** [comment]
Sorry, I don't quite understand the issue. I checked the code and it seems to be perfectly fine. Changing it to `current := atomic.AddUint64(&rl.allowance, passed*rl.unit))` would be wrong. Could you recreate the problem you are trying to fix with a test?

**Comment by july2993** [comment]
```
package main

import (
    "fmt"
    "time"

    rrl "gopkg.in/bsm/ratelimit.v1"
)

func main() {
    lt := rrl.New(100, time.Second)

    st := time.Now()
    cnt := 0

    for {
        if lt.Limit() {
        } else {
            cnt++
        }   

        if st.Add(time.Second * 10).Before(time.Now()) {
            break
        }   
    }   

    fmt.Println(cnt)
}
```

for this cnt printed out will be 109, this is not what we want...

**Comment by dim** [comment]
You are right, fixed in #fbc2d5a2daac4dc3801ee3073874800e2be6ccd1. Thanks
